### DISCUSSION TITLE
Adding new client-specific translation strings workflow

### DISCUSSION THREAD

**Discussion by SchiavoAnto** [ppy/osu#30201] [Q&A] [answered]
While recently working on the Skin Editor, I noticed that many of the strings used in its labels and context menus are using hard-coded English text. This is the case for many other client-specific text elements.

An example from the Skin Editor interface (language is Italian):
![image](https://github.com/user-attachments/assets/445b88f0-0e41-415c-90c3-7de59bd3c67c)

Browsing the issues and PRs, I found [this comment](https://github.com/ppy/osu/issues/4757#issuecomment-862133708) from smoogipoo saying to use the Roslyn tool to create new translatable keys and the only other documentation I could find is [here](https://github.com/ppy/osu/wiki/Localising-the-client) in osu!'s GitHub wiki, but there is no info on how to add NEW translation keys.

I am unsure of the workflow to follow, if it is only the one specified in smoogi's comment of if there is some other procedure (which I know there is for strings shared with osu-web).

**Comment by Joehuu** [comment, marked as answer]
Made an initial writing: https://github.com/ppy/osu/wiki/Localising-the-client#localising-hardcoded-strings
